src/aatoolbox/datasources/icpac/icpac.py

```diff
@@ -0,0 +1,93 @@
+"""Retrieve data from ICPAC's FTP server."""
+import os
+from ftplib import FTP, FTP_TLS
+
+# TODO: decide we would like to use dotenv or not
+from dotenv import load_dotenv
+
+
+def connect_icpac_ftp(
+    ftp_address: str,
+    ftp_username: str,
+    ftp_password: str,
+) -> FTP_TLS:
+    """
+    Connect to ICPAC's ftp.
+
+    To connect you need credentials.
+
+    Parameters
+    ----------
+    ftp_address : str
+        IP address of the ftp server
+    ftp_username : str
+        username of the ftp server
+    ftp_password : str
+        password of the ftp server
+
+    Returns
+    -------
+    ftps: FTP
+        a FTP object
+    """
+    # for some reason it was impossible to access the FTP
+    # this class fixes it, which is copied from this SO
+    # https://stackoverflow.com/questions/14659154/ftps-with-python-ftplib-session-reuse-required
+    class MyFTP_TLS(FTP_TLS):
+        """Explicit FTPS, with shared TLS session."""
+
+        def ntransfercmd(self, cmd, rest=None):
+            conn, size = FTP.ntransfercmd(self=self, cmd=cmd, rest=rest)
+            if self._prot_p:
+                conn = self.context.wrap_socket(
+                    sock=conn,
+                    server_hostname=self.host,
+                    session=self.sock.session,
+                )  # this is the fix
+            return conn, size
+
+    ftps = MyFTP_TLS(host=ftp_address, user=ftp_username, passwd=ftp_password)
+    ftps.prot_p()
+
+    return ftps
+
+
+def retrieve_file_ftp(
+    ftp_filepath: str,
+    output_filepath: str,
+):
+    """
+    Download and save a file from ICPAC's ftp server.
+
+    Parameters
+    ----------
+    ftp_filepath : str
+        path on the server where the file is located
+    output_filepath : str
+        path to save the file to
+
+    Examples
+    --------
+    >>> retrieve_file_ftp(ftp_filepath=
+    ... '/SharedData/gcm/seasonal/202101/' \
+    ... 'PredictedRainfallProbbability-FMA2021_Jan2021.nc',
+    ... output_filepath='example.nc')
+    """
+    load_dotenv()
```

Comment:
Should we be using `find_dotenv()` as well here as per [this SO post](https://stackoverflow.com/questions/41546883/what-is-the-use-of-python-dotenv)? Where does `load_dotenv()` on its own expect to find the file?

Reply:
Reading the documentation of [`load_dotenv()`](https://github.com/theskumar/python-dotenv/blob/45848bb780c26ef0adf7898656f7d3d3f4e2d8ae/src/dotenv/main.py), there is no warning if the file isn't found by default, so might want to be careful here. However, by default, the code also shows that `find_dotenv()` is used, so hopefully shouldn't be too problematic?